### DISCUSSION TITLE
CORS Problem

### DISCUSSION THREAD

**Discussion by maratime** [weather-gov/api#739] [General]
Hello, I can't figure out how to get my code to work in production. I've change the .htaccess file on the server and tried everything I could find on the internet for a fix. Lots of testing......  I need help? 
Works fine in Firefox with Cors Everywhere extension.

Console Error: 
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://aviationweather.gov/api/data/metar?ids=Kjka%2CKcqf%2CKmob&format=geojson. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing). Status code: 200.

My Code:
const url = 'https://aviationweather.gov/api/data/metar?ids=Kjka%2CKcqf%2CKmob&format=geojson'; 

// get weather information
const getWeather = async () => {
  
    const response = await fetch(url, 
      {method: 'GET',
        headers: {
        'Accept': "application/json",
        'User-Agent': "collierairpark.com; wy@wyoung-dev.com",
        'Access-Control-Allow-Origin': "Origin, X-Requested-With, Content-Type, Accept"
      }});
    if(response.status !== 200){
        throw new Error('cannot fetch the data not 200');
      }
    const data = await response.json();
  
    return data;
    
  };
  
  getWeather().then(data => {
    console.log('promise 1 resolved:', data);
      
      if (typeof data.features[3] === "undefined") {
        console.log("features 3 is undefined");
        
        data.features[3] = '';
        if (typeof data.features[2] === "undefined") {
          data.features[2] = '';
         
        }
        if (typeof data.features[1] === "undefined") {
          data.features[1] = '';
        
        }
        
      } else {
        console.log("myVariable is defined");
      }
            
    var metar = data.features[1].properties;
    var metar2 = data.features[2].properties;
    var metar3 = data.features[3].properties;
    
    document.getElementById('icao1').innerText = metar.id;
    document.getElementById('f_cat1').innerText = metar.fltcat;
    document.getElementById('met1').innerText = metar.rawOb;
    document.getElementById('icao2').innerText = metar2.id;
    document.getElementById('f_cat2').innerText = metar2.fltcat;
    document.getElementById('met2').innerText = metar2.rawOb;
    document.getElementById('icao3').innerText = metar3.id;
    document.getElementById('f_cat3').innerText = metar3.fltcat;
    document.getElementById('met3').innerText = metar3.rawOb;
      
  }).catch(err => console.log('rejected:', err.message));

**Comment by scadergit** [comment]
Sorry, this support resource is for api.weather.gov. Please see the [Aviation API](https://aviationweather.gov/data/api/) for support for that service.

**Comment by maratime** [comment]
Somehow you guys are related. I found links to weather.gov from avaitionweather.com and vice versa.

**Reply by scadergit** [reply]
Both are under the National Weather Service.

**Reply by maratime** [reply]
I can get the weather fine form weather.gov using same code, but get cors error from avaitionweather.com. 
Maybe the JSONLD from weather.gov. is why there's no cors error.  You may know?
